### DISCUSSION TITLE
Abstract Factory

### DISCUSSION THREAD

**Discussion by sigmarsson** [dadhi/DryIoc#383] [Q&A]
Good Day !

Perhaps it is already existing. 

Is this container capable to provide a factory implementation out of a given interface ?

E.g. `kernel.Bind<IAbstractFactory>().ToAbstract();` where 

```
interface IAbstractFactory
{
  ViewModelA GetDialogA();
  ViewModelB GetDialogB();
}
```

So you just need to maintain a single interface and not a batch of bindings.

**Comment by dadhi** [comment]
@sigmarsson Hello. If you mean the interface (factory) proxy generation then no.

**Reply by sigmarsson** [reply]
Yeah, I mean the binding above would furnish the implementation of that interface, on the fly.

**Reply by dadhi** [reply]
@sigmarsson I understood, but it requires the generation of the factory class, which is outside of the current DryIoc facilities.
I specifically did not want to overcomplicate the internals with more stuff than needed especially in this case, where you need to add the class generation in order to support some small feature. Which is 99% effort vs 1% of the value.

But probably it may be solved by other tools like `Castle.DynamicProxy` and be used together with the DryIoc to register the factory proxy methods as DryIoc [factory methods](https://github.com/dadhi/DryIoc/blob/master/docs/DryIoc.Docs/SelectConstructorOrFactoryMethod.md#factory-method-instead-of-constructor). The problem is a bit similar to how you may approach the method interception with DryIoc - https://github.com/dadhi/DryIoc/blob/master/docs/DryIoc.Docs/Interception.md

**Reply by sigmarsson** [reply]
Howdy @dadhi , I just though it ought to be relative simple. e.g. furnish the instance of those classes having a default ctor. Only the ones having default. 
I have seen this first [here](https://github.com/canton7/Stylet/blob/5e97f9e1ea8c45e9d7033e6e7c7ea8ce41c3420a/Stylet/StyletIoC/Internal/Builders/BuilderBindTo.cs#L75)
